### ISSUE TITLE
archive: memory allocation of 16378232832 bytes failed

### ISSUE THREAD

**Issue by leo-lb** [m4b/goblin#133]
Hello,

I don't really have time now to debug this but I'd still like to share the test file.
It's being parsed as an archive.

The application exits after printing `memory allocation of 16378232832 bytes failed`.

File: https://github.com/leo-lb/random-files/blob/master/jvm.lib

Thanks.

**Comment by jjunqueira** [comment]
This is happening to me as well, it unfortunately causes an abort panic and crashes my program. I can't figure out a way to catch the error and keep processing. I wonder if its the same issue from here [https://github.com/m4b/goblin/issues/120](url) but it is happening to me on PE files.

**Comment by m4b** [comment]
@leo-lb thanks for the file; so on current master i get:
```
type is too big (4) for 3
```
when i run:
```
cargo run --example=rdr ~/Downloads/jvm.lib
```

`nm` can parse it though without failing so we should do better.

@jjunqueira the issue you linked seems to be a deadlink, but you are also getting OOM errors?

**Comment by jjunqueira** [comment]
@m4b Oops sorry https://github.com/m4b/goblin/issues/120

I am using 0.0.22 so maybe I should use master? 

The part in issue 120 that I think is applying here is "Memory allocator runs out of virtual memory and the process is aborted.".  I am gathering pe file information from large batches of binaries and when I hit this error my whole process is aborted. In my case I would prefer it return an error that I could handle instead of abort.

I'll try doing a test on master and report back!

**Comment by m4b** [comment]
Yes we should never oom or panic, user error should always be returned. This is a bug for sure.

**Comment by m4b** [comment]
Can you upload the offending PE or is it sensitive ? Would like to poke at it and see what’s wrong.

**Comment by jjunqueira** [comment]
@m4b Sorry got sidetracked on a few things and haven't made it back to this yet. I'll ask to see if I can share the sample. I also haven't gotten around to trying it on the master branch yet. I am hoping to get to it in the morning.

**Comment by jjunqueira** [comment]
@m4b @leo-lb So I feel kinda dumb now but I just saw this in the README "Goblin requires rustc 1.31.1." and of course my version is:

▶ rustc --version
rustc 1.34.1 (fc50f328b 2019-04-24)

Then I was thinking that a major pretty recent change was moving away from jemalloc as the default allocator and I am having an abort on memory allocation. So I switched my project to include jemalloc and BOOM problem gone.

@m4b Can you think of any other issues that may come up by using 1.34.1 instead of 1.31.1? I don't mind keeping jemalloc in my project.

Thanks again for your help!

**Comment by philipc** [comment]
@jjunqueira "Goblin requires rustc 1.31.1." means 1.31.1 or newer. It definitely should work with 1.34.1, and we'd prefer to track down the problem rather than having you work around it by using jemalloc. So it'd still be great if you can give us a way to reproduce your error.

**Comment by jjunqueira** [comment]
@philipc @m4b My company does large scale processing of malware so the binary could very well be malware. I can send an encrypted zip file to you directly so its not uploaded here where others could stumble upon it. It may be benign but I just want to be careful. Just let me know what your preferred method is for me to send it to you.

Also for reference it works as expected on my OSX laptop and in a docker container. It is only failing when I deploy it to a CentOS VM that is a bit memory constrained:

`
[dev@dev]# free -h
              total        used        free      shared  buff/cache   available
Mem:           1.8G        417M        433M         88M        977M        1.1G
Swap:          1.0G         28M        995M
[dev@dev]# cat /etc/redhat-release
CentOS Linux release 7.6.1810 (Core)
`

**Comment by raindev** [comment]
@philipc this issue is fixed by e3c5d58 as well but I messed up the commit message so it wasn't closed automatically.

**Comment by philipc** [comment]
Fixed by #174.

@jjunqueira I'm not sure if #174 also fixed the bug you found, but please open a new issue if it didn't.
